ood-portal-generator/lib/ood_portal_generator/view.rb

```diff
@@ -131,13 +132,31 @@ def log_filename(value,log_type)
 
     def allowed_hosts
       config_hosts = []
-      config_hosts << @servername unless @servername.nil?
+
+      add_servername_or_ips(config_hosts)
       config_hosts << @proxy_server unless @proxy_server.nil?
       config_hosts.concat(@server_aliases)
+
       return nil if config_hosts.empty?
+
       config_hosts.sort.uniq
     end
 
+    def add_servername_or_ips(config_hosts)
+      # if @servername is nil, they're trying to use ip addresses
+      if @servername.nil?
+        config_hosts.concat(ip_addresses)
+      else
+        config_hosts << @servername
+      end
+    end
+
+    def ip_addresses
+      Socket.ip_address_list.select(&:ipv4?)
```

Comment:
It's not clear to me if we should allow ipv6 addresses.

What's more is, if I do enable ipv6 address - I'm not really sure how to use them. They seem to have the interface name appended to it, so I'm not entirely sure if we need to cut that off or what.